### PR TITLE
feat(oui-search): add oui-search component

### PR DESCRIPTION
**Pull request by AxelPeter**
UU-568
http://build_oui_search.ui-kit.ovh/#!/oui-angular/search

Linked to
https://github.com/ovh-ux/ovh-ui-kit/pull/158
https://github.com/ovh-ux/ovh-ui-kit-documentation/pull/61

### CODE REVIEW

File: packages/oui-search/src/search.controller.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        if (angular.isFunction(this.onSubmit)) {
+            this.onSubmit({ modelValue });
+        }
+    }
+
+    onSearchReset () {
+        // Since type="reset" doesn't reset the model
+        this.model = undefined;
+
+        if (angular.isFunction(this.onReset)) {
```
Comment (tremblaymath) [inline packages/oui-search/src/search.controller.js]: (optional) If you replace binding from onReset: "&?" to onReset: "&" you will be able to delete  this condition.

File: packages/oui-search/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+| Attribute     | Type     | Binding | One-time Binding | Values    | Default   | Description                   |
+| ----          | ----     | ----    | ----             | ----      | ----      | ----                          |
+| model         | object   | =       |                  |           |           | model bound to component      |
+| id            | string   | @?      | true             |           |           | id attribute of the button    |
+| name          | string   | @?      | true             |           |           | name attribute of the button  |
+| placeholder   | string   | @?      | true             |           |           | placeholder text              |
+| aria-label    | string   | @?      | true             |           |           | accessibility label           |
+| disabled      | boolean  | <?      |                  |           | false     | disabled flag                 |
+| on-change     | function | &?      |                  |           |           | click handler                 |
```
[the remark this answers is not in the document]
Comment (AxelPeter) [reply packages/oui-search/README.md]: Bad copy/paste, i update it

File: packages/oui-search/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    placeholder="..."
+    aria-label="..."
+    disabled|disabled="true|false"
+    on-change="..."
+    on-reset="..."
+    on-submit="...">
+</oui-search>
+```
+
+## API
```
[the remark this answers is not in the document]
Comment (AxelPeter) [reply packages/oui-search/README.md]: Updated

File: packages/oui-search/src/search.controller.js (hunk 1 above)
Comment (AxelPeter) [reply packages/oui-search/src/search.controller.js]: Ok, i'll do for the others too :)

**Pull request merged**